fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/query/BaseWhereAdapter.java

```diff
@@ -0,0 +1,1137 @@
+    }
+
+    /**
+     * Add the string expression directly to the statement
+     * @param expression
+     * @return
+     */
+    public T expr(String expression) {
+        predicateParser.column(expression);
+        return getThis();
+    }
+
+
+    // ### NOT EQUAL ###
+    /**
+     * Add != to the expression
+     * @return
+     */
+    public T neq() {
+        predicateParser.neq();
+        return getThis();
+    }
+
+    /**
+     * Add != {ref} to the expression
+     * @param ref the column reference
+     * @return
+     */
+    public T neq(ColumnRef ref) {
+        predicateParser.neq();
+        predicateParser.column(ref.toString());
+        return getThis();
+    }
+
+    /**
+     * Add != {literalValue} to the expression
+     * @param literalValue
+     * @return
+     */
+    public T neq(long literalValue) {
+        predicateParser.neq();
+        predicateParser.literal(literalValue);
+        return getThis();
+    }
+
+    /**
+     * Add != {literalValue} to the expression
+     * @param literalValue
+     * @return
+     */
+    public T neq(int literalValue) {
+        predicateParser.neq();
+        predicateParser.literal((long)literalValue);
+        return getThis();
+    }
+
+    /**
+     * Add != {literalValue} to the expression
+     * @param literalValue
+     * @return
+     */
+    public T neq(String literalValue) {
+        predicateParser.neq();
+        predicateParser.column(literalValue);
+        return getThis();
+    }
+
+    /**
+     * Add != {str} to the expression
+     * @param str a literal string value
+     * @return
+     */
+    public T neq(LiteralString str) {
+        predicateParser.neq();
+        predicateParser.literal(str.getValue());
+        return getThis();
+    }
+
+    /**
+     * Add a bind marker and its value to the expression
+     * @param bindMarker
+     * @return
+     */
+    public T neq(BindMarkerNode bindMarker) {
+        predicateParser.neq();
+        predicateParser.bindMarker(bindMarker);
+        return getThis();
+    }
+
+    // ### GREATER THAN ###
+    /**
+     * Add > to the expression
+     * @return
+     */
+    public T gt() {
+        predicateParser.gt();
+        return getThis();
+    }
+
+    /**
+     * Add > {ref} to the expression
+     * @param ref the column reference
+     * @return
+     */
+    public T gt(ColumnRef ref) {
+        predicateParser.gt();
+        predicateParser.column(ref.toString());
+        return getThis();
+    }
+
+    /**
+     * Add > {literalValue} to the expression
+     * @param literalValue
+     * @return
+     */
+    public T gt(long literalValue) {
+        predicateParser.gt();
+        predicateParser.literal(literalValue);
+        return getThis();
+    }
+
+    /**
+     * Add > {literalValue} to the expression
+     * @param literalValue
+     * @return
+     */
+    public T gt(int literalValue) {
+        predicateParser.gt();
+        predicateParser.literal((long)literalValue);
```

Comment:
I think I get what is happening... any reason why it's important to cast to long?

Reply:
same question on 520

Reply:
also wouldn't it be easier to pass the cast long into gte(long literalValue) and then use the same code path?

Reply:
similar patterns in the same class with lt/gt lte/gte

Reply:
There's no IntExpNode, only LongExpNode and the PredicateParser literal method takes a Long not a long. So the cast is required to permit boxing to work. We could introduce an IntExpNode and associated methods if needed.

Reply:
I don't think it's  needed.  just felt a little strange.